test/suites/smoke/test-balances-consistency.ts

```diff
@@ -356,312 +322,88 @@ describeSuite({
       });
 
       await new Promise((resolve, reject) => {
-        if (specVersion < 2900) {
-          apiAt.query.democracy.depositOf
-            .entries()
-            .then((democracyDeposits) => {
-              democracyDeposits
-                .map((depositOf) =>
-                  depositOf[1]
-                    .unwrap()[0]
-                    .flatMap((deposit) => ({
-                      accountId: deposit.toHex(),
-                      reserved: depositOf[1].unwrap()[1].toBigInt(),
-                    }))
-                    .reduce(
-                      (p, deposit) => {
-                        // We merge multiple reserves together for same account
-                        if (!p[deposit.accountId]) {
-                          p[deposit.accountId] = {
-                            accountId: deposit.accountId,
-                            reserved: {
-                              [ReserveType.DemocracyDeposit]: 0n,
-                            },
-                          };
-                        }
-                        p[deposit.accountId].reserved[ReserveType.DemocracyDeposit] +=
-                          deposit.reserved;
-                        return p;
-                      },
-                      {} as {
-                        [account: string]: {
-                          accountId: `0x${string}`;
-                          reserved: {
-                            [ReserveType.DemocracyDeposit]: bigint;
-                          };
-                        };
-                      }
-                    )
-                )
-                .forEach((deposit: any) => {
-                  updateReserveMap(deposit.accountId, deposit.reserved);
+        apiAt.query.preimage.statusFor
+          .entries()
+          .then((preimageStatuses) => {
+            preimageStatuses
+              .filter(
+                (status) => status[1].unwrap().isUnrequested || status[1].unwrap().isRequested
+              )
+              .map((status) => {
+                const deposit = extractPreimageDeposit(
+                  status[1].unwrap().isUnrequested
+                    ? status[1].unwrap().asUnrequested
+                    : status[1].unwrap().asRequested
+                );
+                return deposit
+                  ? { accountId: deposit.accountId, deposit: deposit.amount }
+                  : undefined;
+              })
+              .filter((value) => typeof value !== "undefined")
+              .forEach(({ deposit, accountId }: any) => {
+                updateReserveMap(accountId, {
+                  [ReserveType.PreimageStatus]: deposit === 0n ? 0n : deposit.toBigInt(),
                 });
```

Comment:
nit: Can we do `status[1].unwrap()` once at the beginning and have something like this:

```suggestion
preimageStatuses
    .map((status) => status[1].unwrap())
    .filter((unwrapped) => unwrapped.isUnrequested || unwrapped.isRequested)
    .map((unwrapped) => {
      const statusData = unwrapped.isUnrequested ? unwrapped.asUnrequested : unwrapped.asRequested;
      return extractPreimageDeposit(statusData);
    })
    .filter((deposit): deposit is NonNullable<typeof deposit> => deposit !== undefined)
    .forEach(({ accountId, amount }) => {
      updateReserveMap(accountId, {
        [ReserveType.PreimageStatus]: amount === 0n ? 0n : amount.toBigInt(),
      });
    });
});
```

Reply:
This code already existed. I agree that a lot of things (not only specific to this test file) can be simplified, I think we already have a task on Jira for that.

Reply:
Sorry, I thought it was new code 👍